genie-core/src/main/java/com/netflix/genie/core/jobs/workflow/impl/JobKickoffTask.java

```diff
@@ -107,6 +108,12 @@ public void executeTask(
             command.add("-u");
             command.add(this.jobExecEnv.getJobRequest().getUser());
         }
+
+        // If the OS is linux use setsid to launch the process so that the entire process tree
+        // is launched in process group id which is the same as the pid of the parent process
+        if (SystemUtils.IS_OS_LINUX) {
+            command.add("setsid");
+        }
```

Comment:
If we're going to start having lots of OS level differences I'd say we should create subclasses for LinuxJobKickoffTask, OSXJobKickoffTask, etc... that are created as beans based on what OS it is. That may well be overkill for right now but it's something to keep in mind.